simulation/halsim_ws_core/src/main/native/cpp/WSProvider_DriverStation.cpp

```diff
@@ -0,0 +1,174 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017-2020 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "WSProvider_DriverStation.h"
+
+#include <algorithm>
+
+#include <hal/DriverStation.h>
+#include <hal/Ports.h>
+#include <hal/simulation/DriverStationData.h>
+#include <wpi/raw_ostream.h>
+
+#define REGISTER(halsim, jsonid, ctype, haltype)                          \
+  HALSIM_RegisterDriverStation##halsim##Callback(                         \
+      [](const char* name, void* param, const struct HAL_Value* value) {  \
+        static_cast<HALSimWSProviderDriverStation*>(param)                \
+            ->ProcessHalCallback(                                         \
+                {{jsonid, static_cast<ctype>(value->data.v_##haltype)}}); \
+      },                                                                  \
+      this, true)
+
+namespace wpilibws {
+
+void HALSimWSProviderDriverStation::Initialize(WSRegisterFunc webRegisterFunc) {
+  // TODO: if this isn't called, our callbacks get erased. Fixed in 2021
```

Comment:
Is this comment still true?

Reply:
Just tried it without the initialization call and it seemed to work fine